src/Factory.php

```diff
@@ -68,18 +69,43 @@ public function add(string $set, array $options): self
     public function registerComponents(): void
     {
         foreach ($this->sets as $set) {
-            foreach ($this->filesystem->allFiles($set['path']) as $file) {
-                $path = array_filter(explode('/', Str::after($file->getPath(), $set['path'])));
-
-                Blade::component(
-                    SvgComponent::class,
-                    implode('.', array_filter($path + [$file->getFilenameWithoutExtension()])),
-                    $set['prefix']
-                );
+            if (isset($set['path'])) {
+                $this->registerComponentsByPath($set);
+            } else {
+                $this->registerComponentsByDisk($set);
```

Comment:
I'd inverse this and make `path` the default one. That also makes the upgrade path for all first party libraries graceful.

Reply:
This case always preference is 'path'
If same time in set given path, and disk then all code will work as before. Disk will be ignore that case